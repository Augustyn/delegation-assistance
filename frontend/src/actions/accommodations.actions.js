import { APIService } from "../services/data";

export const ACTIONS = {
  FETCH_ACCOMMODATIONS: "ACCOMMODATIONS_FETCH_ACCOMMODATIONS",
  ADD_ACCOMMODATION: "ACCOMMODATIONS_ADD_ACCOMMODATION",
  SORT_ACCOMMODATIONS: "ACCOMMODATIONS_SORT_ACCOMMODATIONS"
};

const addAccommodation = (delegationId, accommodation) => dispatch => {
  return dispatch(
    APIService.post(ACTIONS.ADD_ACCOMMODATION, {
      url: `/delegations/${delegationId}/accommodations`,
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: accommodation
    })
  );
};

const fetchAccommodations = delegationId => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.FETCH_ACCOMMODATIONS, {
      url: `/delegations/${delegationId}/accommodations`,
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const sortAccommodationsByCheckInDate = order => dispatch => {
  return dispatch({
    type: ACTIONS.SORT_ACCOMMODATIONS,
    payload: order
  });
};

export { addAccommodation, fetchAccommodations, sortAccommodationsByCheckInDate };
