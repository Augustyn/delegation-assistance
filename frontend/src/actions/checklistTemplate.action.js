import { APIService } from "../services/data";

export const ACTIONS = {
  GET_GLOBAL_TEMPLATE: "CHECKLIST_TEMPLATE_GET_GLOBAL_TEMPLATE",
  REORDER_GLOBAL_TEMPLATE: "CHECKLIST_TEMPLATE_REORDER_GLOBAL_TEMPLATE",
  SAVE_GLOBAL_TEMPLATE: "CHECKLIST_TEMPLATE_SAVE_GLOBAL_TEMPLATE",
  ADD_TASK: "CHECKLIST_TEMPLATE_ADD_TASK",
  EDIT_TASK: "CHECKLIST_TEMPLATE_EDIT_TASK",
  DELETE_TASK: "CHECKLIST_TEMPLATE_DELETE_TASK"
};

const fetchGlobalTemplate = () => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.GET_GLOBAL_TEMPLATE, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const saveGlobalTemplate = globalTemplate => dispatch => {
  return dispatch(
    APIService.put(ACTIONS.SAVE_GLOBAL_TEMPLATE, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: {
        activities: [...globalTemplate]
      }
    })
  );
};

const reorderGlobalTemplate = ({ oldIndex, newIndex }) => dispatch =>
  dispatch({
    type: ACTIONS.REORDER_GLOBAL_TEMPLATE,
    payload: {
      oldIndex: oldIndex,
      newIndex: newIndex
    }
  });

const addTask = ({ task, description }) => dispatch =>
  dispatch({
    type: ACTIONS.ADD_TASK,
    payload: {
      task: task,
      description: description
    }
  });

const editTask = ({ id, version, task, description, priority }) => dispatch =>
  dispatch({
    type: ACTIONS.EDIT_TASK,
    payload: {
      id: id,
      version: version,
      task: task,
      description: description,
      priority: priority
    }
  });

const deleteTask = index => dispatch =>
  dispatch({
    type: ACTIONS.DELETE_TASK,
    payload: {
      index: index
    }
  });

export { fetchGlobalTemplate, reorderGlobalTemplate, saveGlobalTemplate, addTask, editTask, deleteTask };
