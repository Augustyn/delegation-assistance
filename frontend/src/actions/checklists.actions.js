import { APIService } from "../services/data";

export const ACTIONS = {
  ADD_TASK: "CHECKLISTS_ADD_TASK",
  GET_TASKS: "CHECKLISTS_GET_TASKS",
  DELETE_TASK: "CHECKLISTS_DELETE_TASK"
};

const addTask = (name, description) => dispatch => {
  return dispatch(
    APIService.post(
      ACTIONS.ADD_TASK,
      {
        url: "/checklist/tasks",
        headers: {
          "Content-type": "application/json"
        },
        needAuth: true,
        data: {
          name: name,
          description: description
        }
      },
      {
        name: name,
        description: description
      }
    )
  );
};

const deleteTask = id => dispatch => {
  let url = "/checklist/tasks/" + id;
  return dispatch(
    APIService.delete(
      ACTIONS.DELETE_TASK,
      {
        url: url,
        headers: {
          "Content-type": "application/json"
        },
        needAuth: true
      },
      {
        id: id
      }
    )
  );
};

const fetchTasks = () => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.GET_TASKS, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const saveTasks = tasks => dispatch => {
  return dispatch(
    APIService.put(ACTIONS.GET_TASKS, {
      url: "/checklist",
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: {
        tasks: tasks
      }
    })
  );
};

export { addTask, deleteTask, fetchTasks, saveTasks };
