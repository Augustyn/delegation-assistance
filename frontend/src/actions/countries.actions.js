import { APIService } from "../services/data";

export const ACTIONS = {
  FETCH_COUNTRIES: "COUNTRIES_FETCH_COUNTRIES"
};

const fetchCountries = () => (dispatch, getState) => {
  // Fetching only when coutries collection is empty in redux store.
  const countries = getState().countries.countries;
  if (countries.length === 0) {
    return dispatch(
      APIService.get(ACTIONS.FETCH_COUNTRIES, {
        url: `/countries`,
        headers: {
          "Content-type": "application/json"
        },
        needAuth: true
      })
    );
  }
  return Promise.resolve();
};

export { fetchCountries };
