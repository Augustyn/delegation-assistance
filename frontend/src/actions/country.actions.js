import { RESTCountries } from "../services/data";

export const ACTIONS = {
  FETCH_COUNTRY: "COUNTRIES_FETCH_COUNTRY"
};

const fetchCountry = countryCode => dispatch => {
  return dispatch(
    RESTCountries.get(ACTIONS.FETCH_COUNTRY, {
      url: `/alpha/${countryCode}`,
      headers: {
        "Content-type": "application/json"
      }
    })
  );
};

export { fetchCountry };
