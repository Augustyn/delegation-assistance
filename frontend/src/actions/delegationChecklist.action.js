import { APIService } from "../services/data";

export const ACTIONS = {
  FETCH_DELEGATION_CHECKLIST: "DELEGATION_CHECKLIST_FETCH_DELEGATION_CHECKLIST",
  UPDATE_DELEGATION_CHECKLIST: "DELEGATION_CHECKLIST_UPDATE_DELEGATION_CHECKLIST"
};

const fetchDelegationChecklist = delegationId => dispatch => {
  return dispatch(
    APIService.get(ACTIONS.FETCH_DELEGATION_CHECKLIST, {
      url: `/delegations/${delegationId}/checklist`,
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true
    })
  );
};

const updateDelegationChecklist = (delegationId, checklist) => dispatch => {
  return dispatch(
    APIService.patch(ACTIONS.UPDATE_DELEGATION_CHECKLIST, {
      url: `/delegations/${delegationId}/checklist`,
      headers: {
        "Content-type": "application/json"
      },
      needAuth: true,
      data: checklist
    })
  );
};

export { fetchDelegationChecklist, updateDelegationChecklist };
