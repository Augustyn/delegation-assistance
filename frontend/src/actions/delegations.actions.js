import {APIService} from "../services/data";

export const ACTIONS = {
  POST_DELEGATION: "DELEGATIONS_POST_DELEGATION"
};

const addNewDelegation = (delegation) => dispatch => {
  delegation.startDate = delegation.startDate.toISOString();
  delegation.endDate = delegation.endDate.toISOString();
  return dispatch(
    APIService.post(ACTIONS.POST_DELEGATION, {
      url: '/delegations',
      //TODO: add authorization token from sessionStorage(?)
      headers: {
        "Content-type": "application/json",
        "Bearer": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJlbXBsb3llZSIsIm5iZiI6MTU1Mjg1Njg4NSwiaXNzIjoiaWRlbWlhIiwiZXhwIjoxNTUyODYwNDg1LCJpYXQiOjE1NTI4NTY4ODUsImF1dGhvcml0aWVzIjpbIlJPTEVfRU1QTE9ZRUUiXX0.PINY8fxa2TKFut7UBf-S08eKCZ7Yo2FolyU6AjwqpfzK3mSRIvlEHJUgejTq86lImFDhm_J5lb34KQbIhaHaEA"
      },
      needAuth: false,
      data: delegation
    })
  );
};

export {
  addNewDelegation
};
