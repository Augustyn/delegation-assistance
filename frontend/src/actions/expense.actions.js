import { APIService } from "../services/data";

export const ACTIONS = {
  ADD_EXPENSE: "DELEGATIONS_ADD_EXPENSE"
};

const addExpense = (expense, delegationId) => dispatch => {
  return dispatch(
    APIService.post(ACTIONS.ADD_EXPENSE, {
      url: `/delegations/${delegationId}/expenses`,
      headers: {
        "Content-type": "multipart/form-data"
      },
      needAuth: true,
      data: expense
    })
  );
};

export { addExpense };
