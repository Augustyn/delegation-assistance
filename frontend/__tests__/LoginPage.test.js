import React from "react";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { LoginPage } from "../src/features/LoginPage/LoginPage.component";
import { loginUser } from "../src/actions/user.actions";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const props = {
    handleSubmit: jest.fn(),
    subbmiting: false,
    invalid: false
  };

  const enzymeWrapper = shallow(<LoginPage {...props} />);

  return {
    props,
    enzymeWrapper
  };
};

describe("Create Login Page", () => {
  it("should render correctly", () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it("should call submit with user object", () => {
    const value = {
      login: "employee",
      password: "pass1"
    };
    const { enzymeWrapper, props } = setup();
    expect(props.handleSubmit).toHaveBeenCalledTimes(0);
    enzymeWrapper.find("Form").simulate("submit", value);
    expect(props.handleSubmit).toBeCalledWith(value);
  });

  it("should render 2 Inputs", () => {
    const { enzymeWrapper } = setup();
    expect(enzymeWrapper.find("Input").length).toBe(2);
  });

  it("should dispatch action", () => {
    const initialState = {};
    const store = mockStore(initialState);
    store.dispatch(loginUser());

    const actions = store.getActions();
    const expectedPayload = "USER_LOGIN_USER";
    expect(actions[0].type).toEqual(expectedPayload);
  });
});
