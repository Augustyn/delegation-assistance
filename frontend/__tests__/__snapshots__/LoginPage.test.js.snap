// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Login Page should render correctly 1`] = `
<Connect(LayoutMainContainer)
  fullContent={true}
  hideTitle={true}
  title="Sign in"
>
  <div
    id="login-screen"
    style={
      Object {
        "backgroundImage": "url(test-file-stub)",
      }
    }
  >
    <div
      className="login-container"
    >
      <h1
        className="title-text"
      >
        Nice to see you again!
      </h1>
      <Form
        className="form-container"
        onSubmit={[MockFunction]}
      >
        <span
          style={
            Object {
              "color": "red",
            }
          }
        />
        <Input
          component="input"
          label="Login *"
          name="login"
          placeholder="Username"
          validate={[Function]}
        />
        <Input
          component="input"
          label="Password *"
          name="password"
          placeholder="Password"
          type="password"
          validate={[Function]}
        />
        <Button
          text="sign in"
          type="submit"
        />
      </Form>
    </div>
  </div>
</Connect(LayoutMainContainer)>
`;
